docs/crds/applicationprofile.md

```diff
@@ -0,0 +1,661 @@
+kubectl describe httproute <name>
+```
+
+**6. ApplicationProfile UUID Not Found**
+*Symptom*: Gateway API controller logs show "ApplicationProfile UUID not found"
+
+*Solution*:
+- Ensure ApplicationProfile has been processed by AKO CRD Operator
+- Check that ApplicationProfile status contains a valid UUID
+- Verify ApplicationProfile tenant matches namespace tenant
+
+```bash
+kubectl get applicationprofile <name> -o jsonpath='{.status.uuid}'
+```
+
+### Debug Commands
+
+```bash
+# Check ApplicationProfile status
+kubectl get applicationprofiles -A
+kubectl describe applicationprofile <name> -n <namespace>
+
+# Check ApplicationProfile events
+kubectl get events --field-selector involvedObject.name=<applicationprofile-name> -n <namespace>
+
+# Check AKO CRD Operator logs
+kubectl logs deployment/ako-crd-operator -n avi-system -f
+
+# Check Gateway API controller logs
+kubectl logs deployment/ako-gateway-api -n avi-system -f
+
+# Check HTTPRoute status and conditions
+kubectl get httproute <name> -o yaml
+kubectl describe httproute <name>
+
+# Verify ApplicationProfile UUID
+kubectl get applicationprofile <name> -o jsonpath='{.status.uuid}'
+
+# Check ApplicationProfile to HTTPRoute mappings
+kubectl logs deployment/ako-gateway-api -n avi-system | grep "ApplicationProfile"
+
+# Verify Avi Controller configuration
+# (Access Avi Controller UI to verify Application Profile configuration)
+```
+
+### Status Interpretation
+
+**Programmed: True**
+- ApplicationProfile has been successfully created on the Avi Controller
+- Ready to be referenced by Virtual Services
+
+**Programmed: False with reason "CreationFailed"**
+- Failed to create ApplicationProfile on Avi Controller
+- Check controller logs for detailed error information
+
+**Programmed: False with reason "UUIDExtractionFailed"**
+- ApplicationProfile was created but UUID extraction failed
+- Check Avi Controller connectivity and permissions
+
+### Validation Rules
+
+The ApplicationProfile CRD enforces the following validation rules:
+
+1. **Type Requirement**: The `type` field is required and must be "HTTP"
+2. **XFF Configuration**: `xff_alternate_name` and `xff_update` can only be configured if `xff_enabled` is true
+3. **XFF Update Constraint**: If `xff_update` is "APPEND_TO_THE_XFF_HEADER", `xff_alternate_name` must be empty
+4. **True Client IP Configuration**: `true_client_ip` can only be configured if `use_true_client_ip` is true
+5. **Header Length**: Each header name in `true_client_ip.headers` must be at most 128 characters long
+6. **Index Range**: `index_in_header` must be between 1 and 1000
+
+### Prerequisites
+
+Before creating an ApplicationProfile CRD:
+
+1. The AKO CRD Operator must be installed and running in the cluster
+2. Ensure the Avi Controller is accessible and properly configured
+3. Verify that the namespace where the ApplicationProfile will be created exists
+
+### Namespace Considerations
+
+- ApplicationProfile resources can be created in any namespace
```

Comment:
As per my understanding, we should just mention common namespace between httproute and app profile. This point invalidates due to second point.